packages/babel-plugin-react-svg/src/index.js

```diff
@@ -146,25 +146,30 @@ export default function(babel: BabelCore) {
     }
   };
 
-  const programVisitor = {
-    Program(path: any) {
-      // add import react statement
-      path.node.body.unshift(
-        t.importDeclaration(
-          [t.importDefaultSpecifier(t.identifier("React"))],
-          t.stringLiteral("react")
-        )
-      );
-    }
-  };
+  const visitor = Object.assign(
+    {},
+    svgExpressionVisitor,
+    svgVisitor,
+    attrVisitor
+  );
 
   return {
-    visitor: Object.assign(
-      {},
-      programVisitor,
-      svgExpressionVisitor,
-      svgVisitor,
-      attrVisitor
-    )
+    visitor: {
+      Program(path: any, state: any) {
+        if (!/\.svg$/.test(state.file.opts.filename)) {
```

Comment:
we could probably make this configurable or use better heuristic, if you have any suggestions how you'd like to have it adjusted, let me know